src/main/java/hsleiden/stenentijdperk/stenentijdperk/Controllers/BoardController.java

```diff
@@ -83,18 +81,20 @@ public void onButtonClick(int index) {
 
     // Hier is het rollen voor resources.
     public void resolveResource(int index) {
-        gegooideWorp[0] = index;
-        int stamleden = playercontroller.getPositie(boardmodel.getPlayer(), index);
-        if (stamleden != 0) {
-            Dobbelsteen roll = new Dobbelsteen(stamleden);
-            roll.worp();
-            roll.berekenTotaal();
-            gegooideWorp[1] = roll.getTotaal();
-            gegooideWorp[2] = stamleden;
-            if (playercontroller.getTools(boardmodel.getPlayer()).size() != 0 && checkTools()) {
-                ViewManager.loadPopupWindow(new TableauView(boardmodel.getPlayer(), this, gegooideWorp[1]).setScene());
-            } else {
-                toolsGebruiken(0);
+        if(FirebaseController.getBoard(String.valueOf(localPlayer.getLobby())).getPlayer() == localPlayer) {
+            gegooideWorp[0] = index;
+            int stamleden = playercontroller.getPositie(boardmodel.getPlayer(), index);
+            if (stamleden != 0) {
+                Dobbelsteen roll = new Dobbelsteen(stamleden);
+                roll.worp();
+                roll.berekenTotaal();
+                gegooideWorp[1] = roll.getTotaal();
+                gegooideWorp[2] = stamleden;
+                if (playercontroller.getTools(boardmodel.getPlayer()).size() != 0 && checkTools()) {
+                    ViewManager.loadPopupWindow(new TableauView(boardmodel.getPlayer(), this, gegooideWorp[1]).setScene());
+                } else {
+                    toolsGebruiken(0);
+                }
```

Comment:
hetzelfde hier. dit is goed, maar dit zal ervoor zorgen dat we heel veel calls maken